src/image.rs

```diff
@@ -14,59 +14,66 @@ impl<'c> TileImage<'c> {
         let width_in_tiles = self.img.width() / self.config.tilesize;
         let height_in_tiles = self.img.height() / self.config.tilesize;
         let morton_idx_max = width_in_tiles * height_in_tiles;
+        let tileimage_coord: (u16, u16) = match &self.config.parentzoomlevel {
+            None => (0, 0),
+            Some(parentzoomlevel) => {
+                if parentzoomlevel > &self.config.zoomlevel {
+                    coord_of(self.config.indexforzoom.into())
+                } else {
+                    (0, 0)
+                }
+            }
+        };
+        let output_level = match &self.config.parentzoomlevel {
+            None => index,
+            Some(parentzoomlevel) => {
+                if parentzoomlevel > &self.config.zoomlevel {
+                    *parentzoomlevel
+                } else {
+                    index
+                }
+            }
+        };
 
         let mut targetrangetoslice = 0..=morton_idx_max - 1;
         // if startzoomrangetoslice is the same as endzoomrangetoslice,
         // then tiles to be sliced in this function are from same zoom level
-        if self.config.startzoomrangetoslice == self.config.endzoomrangetoslice {
-            if index == self.config.endzoomrangetoslice {
-                targetrangetoslice = self.config.starttargetrange..=self.config.endtargetrange;
+        if self.config.zoomrangetoslice.start() == self.config.zoomrangetoslice.end() {
+            if index == *self.config.zoomrangetoslice.end() {
+                targetrangetoslice =
+                    *self.config.targetrangetoslice.start()..=*self.config.targetrangetoslice.end();
             }
         // otherwise, the start zoom level should slice tiles from starttargetrange to end,
         // the end zoom level should slice tiles from 0 to endtargetrange
-        } else if index == self.config.startzoomrangetoslice {
+        } else if index == *self.config.zoomrangetoslice.start() {
             if index > 0 {
-                targetrangetoslice = self.config.starttargetrange..=(1 << (index * 2)) - 1;
+                targetrangetoslice =
+                    *self.config.targetrangetoslice.start()..=(1 << (index * 2)) - 1;
             }
-        } else if index == self.config.endzoomrangetoslice {
-            targetrangetoslice = 0..=self.config.endtargetrange;
+        } else if index == *self.config.zoomrangetoslice.end() {
+            targetrangetoslice = 0..=*self.config.targetrangetoslice.end();
         }
 
         targetrangetoslice
             .map(|morton_idx| {
                 let coord = coord_of(morton_idx);
-                let x = coord.0 as u32;
-                let y = coord.1 as u32;
+                let tilename = format!(
+                    "{z}-{x}-{y}",
+                    z = output_level,
+                    x = (coord.0 + (tileimage_coord.0 * 32)) as u32,
```

Comment:
This hardcoded multiplier assumes that the subimages are always going to be level 5 (32x32 tiles), and fun things happen if you try to use smaller subimages. Not sure how to write this in rust, but what the multiplier should actually be is `2 << zoomlevel`.